libmariadb/mariadb_lib.c

```diff
@@ -3830,6 +3832,9 @@ mysql_optionsv(MYSQL *mysql,enum mysql_option option, ...)
       }
     }
     break;
+  case MARIADB_OPT_BULK_UNIT_RESULTS:
+    mysql->options.bulk_unit_results= *(my_bool *)arg1;
```

Comment:
This will break ABI, since the size of MYSQL will change. Intstead option should be stored in mysql->options.extensions via OPT_SET_EXTENDED_VALUE (and bulk_unit_results should be moved from st_mysql_options to st_mysql_options_extension in ma_common.h)